### DISCUSSION TITLE
Help with text not aligning bottom within layer

### DISCUSSION THREAD

**Discussion by Kazagan** [QuestPDF/QuestPDF#712] [Q&A] [answered]
![Screenshot 2023-11-14 at 3 34 20 PM](https://github.com/QuestPDF/QuestPDF/assets/46536599/c1b8ab5e-a589-43c9-b621-7f8625206306)

```
.Layers(layer =>
                {
                    layer.Layer().Border(.25f)
                        .BorderColor(Colors.Red.Medium)
                        .AlignBottom()
                        .Text(dots)
                        .FontSize(_expandedStatFontSize);
                    layer.PrimaryLayer()
                        .AlignBottom()
                        .MinimalBox()
                        .Background(Colors.White)
                        .Text(t => t.RenderLines(description, _expandedStatFontSize));
                });
```

The dots text is meant to be behind the primary layer, to create a dotted line to connect two points of data. Single line, it works great, but when allowing for some wrapping ( manually managed in the RenderLines method ) the dots are centered instead of at the bottom. I added the red border to show that it recognizes that it has knowledge of the space being taller than that, but it's still not going to the bottom of the area alloted to it.

**Comment by MarcinZiabek** [comment, marked as answer]
Are you sure that `_expandedStatFontSize` does not wrap to the new line and the bottom line is not hidden by the `description` content?

**Reply by Kazagan** [reply]
Definitely should have clarified, the 3 dots after the are an ellipsis to mark truncation, and are not related to the dots I'm attempting to render to connect data. 

And you're correct, it is wrapping to the next line. should have removed that background sooner to check that.

**Reply by MarcinZiabek** [reply]
Great! If the problem is resolved, please feel free to close this discussion 😀

**Reply by Kazagan** [reply]
Is there a way to stop text from wrapping?

**Reply by MarcinZiabek** [reply]
You can use the StopPaging API and then  specify exact height of the inner container, to ensure the text takes only one line.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
